docs/content/design/coordinator.md

```diff
@@ -86,17 +86,30 @@ Once a compact task fails, the coordinator simply finds the segments for the int
 
 #### Newest Segment First Policy
 
-This policy searches the segments of _all dataSources_ in inverse order of their intervals.
-For example, let me assume there are 3 dataSources (`ds1`, `ds2`, `ds3`) and 5 segments (`seg_ds1_2017-10-01_2017-10-02`, `seg_ds1_2017-11-01_2017-11-02`, `seg_ds2_2017-08-01_2017-08-02`, `seg_ds3_2017-07-01_2017-07-02`, `seg_ds3_2017-12-01_2017-12-02`) for those dataSources.
-The segment name indicates its dataSource and interval. The search result of newestSegmentFirstPolicy is [`seg_ds3_2017-12-01_2017-12-02`, `seg_ds1_2017-11-01_2017-11-02`, `seg_ds1_2017-10-01_2017-10-02`, `seg_ds2_2017-08-01_2017-08-02`, `seg_ds3_2017-07-01_2017-07-02`].
-
-Every run, this policy starts searching from the (very latest interval - [skipOffsetFromLatest](../configuration/index.html#compaction-dynamic-configuration)).
-This is to handle the late segments ingested to realtime dataSources.
+At every coordinator run, this policy searches segments to compact by iterating segments from the latest to the oldest.
+Once it finds the latest segment among all dataSources, it checks the segment is _compactible_ with other segments of the same dataSource which have the same or abutting intervals.
+Note that segments are compactible if their total size is smaller than or equal to the configured `inputSegmentSizeBytes`.
+
+Here are some details with example. Let us assume we have two dataSources (`foo`, `bar`)
+and 5 segments (`foo_2017-10-01/2017-11-01`, `foo_2017-11-01/2017-12-01`, `bar_2017-08-01/2017-09-01`, `bar_2017-09-01/2017-10-01`, `bar_2017-10-01/2017-11-01`).
```

Comment:
Would be better for these segments to have a normal looking segment id

Reply:
Or maybe just leave the version off but use a `_` so it's like a prefix of a normal segment id

Reply:
Changed to something like `foo_2017-10-01T00:00:00.000Z_2017-11-01T00:00:00.000Z_VERSION`. Do you think it's better to use a real version string?